packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1188,12 +1193,24 @@ function getValueType(
 	}
 }
 
-export function getDefaultErrorReporter<T>(
+function getErrorRange<T extends AstNode>(
+	node: RuntimeNode<T>,
+	error: McdocCheckerError<T>['kind'],
+): Range {
```

Comment:
I was confused why the build was failing, but this explains it. You're assuming here that `T` will always be an `AstNode`. The runtime checker does not assume that (as is demonstrated by the runtime checker tests, which use simple JS objects). Therefore, a `reportError` options will still need to be required, but you can just simplify those to not check the node type. Reverting this latest commit is probably the easiest.

Reply:
this commit is incomplete, I only pushed it to be able to continue working on it on my laptop 😄